plugin/src/main/java/org/gradle/testretry/internal/framework/TestNgTestFrameworkStrategy.java

```diff
@@ -59,7 +64,19 @@ public TestFramework createRetrying(JvmTestExecutionSpec spec, Test testTask, Se
                 }
             });
 
-        return new TestNGTestFramework(testTask, retriedTestFilter, instantiator, classLoaderCache);
+        VersionNumber gradleVersion = VersionNumber.parse(testTask.getProject().getGradle().getGradleVersion());
+        if (gradleVersion.getMajor() >= 6 && gradleVersion.getMinor()>= 6) {
+            final ObjectFactory objectFactory = ((ProjectInternal) testTask.getProject()).getServices().get(ObjectFactory.class);
+            return new TestNGTestFramework(testTask, objectFactory.fileCollection(), retriedTestFilter, objectFactory);
```

Comment:
I wonder why this doesn't fail any test but we should pass the test classpath:

```suggestion
            return new TestNGTestFramework(testTask, testTask.getClasspath(), retriedTestFilter, objectFactory);
```

Reply:
changed as suggested